src/riak_kv_ts_svc.erl

```diff
@@ -152,7 +152,13 @@ process(M = #riak_sql_describe_v1{'DESCRIBE' = Table}, State) ->
     check_table_and_call(Table, fun sub_tsqueryreq/4, M, State);
 
 process(M = #riak_sql_insert_v1{'INSERT' = Table}, State) ->
-    check_table_and_call(Table, fun sub_tsqueryreq/4, M, State).
+    check_table_and_call(Table, fun sub_tsqueryreq/4, M, State);
+
+process(#riak_sql_show_tables_v1{}, State) ->
+    {ok, {ColNames, ColTypes, LdbNativeRows}} =
+        riak_kv_ts_api:query("SHOW TABLES", ?DDL{}),
```

Comment:
Instead of calling `riak_kv_ts_api:query/2` with a hard-coded SQL statement, you could call `riak_kv_qry:submit(M, ?DDL{})` (where `M` would be the `#riak_sql_show_tables_v1{}` argument matched in function head, as in other `process` heads). By the time execution reaches `riak_kv_ts_svc:process/2`, the original SQL statement will have already been parsed (resulting in `#riak_sql_show_tables_v1{}`, which is why we are here), and there's no need to create a new, identical statement and have it parsed again.